src/effects/effectchainslot.cpp

```diff
@@ -40,6 +40,19 @@ EffectChainSlot::EffectChainSlot(const QString& group,
             ConfigKey(m_group, "num_effectslots"));
     m_pControlNumEffectSlots->setReadOnly();
 
+    m_pControlNumPresetsAvailable = std::make_unique<ControlObject>(
+            ConfigKey(m_group, "num_presetsavailable"));
+    m_pControlNumPresetsAvailable->set(m_pChainPresetManager->numPresets());
+    m_pControlNumPresetsAvailable->setReadOnly();
+    connect(m_pChainPresetManager.get(),
+            &EffectChainPresetManager::effectChainPresetListUpdated,
+            this,
+            &EffectChainSlot::slotPresetListUpdated);
+    connect(m_pChainPresetManager.get(),
+            &EffectChainPresetManager::quickEffectChainPresetListUpdated,
+            this,
+            &EffectChainSlot::slotPresetListUpdated);
```

Comment:
EffectChainPresetManager::effectChainPresetListUpdated should not be connected in QuickEffectChainSlot and EffectChainPresetManager::quickEffectChainPresetListUpdated should *only* be connected in QuickEffectChainSlot

Reply:
I have no idea how to implement this without calling virtual methods from the constructor.

Reply:
You could delete the connection for the QuickEffects here then in the QuickEffectChainSlot constructor, disconnect from EffectChainPresetManager::effectChainPresetListUpdated and connect to EffectChainPresetManager::quickEffectChainPresetListUpdated

Reply:
Tried that here: https://github.com/Holzhaus/mixxx/commit/0faf9c22595e0c55a579e6f0f5383a9a9b3354c9

Unfortunately it doesn't work properly (num_preset is always 0 in quickeffectchainslots). No idea why.

Reply:
The problem was that the quickEffectChainPresetListUpdated signal was not being emitted when effects.xml was empty. I fixed this [here](https://github.com/mixxxdj/mixxx/pull/2618/commits/8503850dfd116f172be8b1e2e9a1cafc2cef18bc#diff-e265e2ac4c841e46240a9480893b1ec3R483) and squashed it together with this PR into one commit.